retro/sadtalker.py

```diff
@@ -39,6 +41,44 @@
 MAX_RES = 1920 * 1080
 
 
+# the original sadtalker function does not work for short audio
+def fixed_generate_blink_seq_randomly(num_frames):
+    ratio = np.zeros((num_frames, 1))
+    if int(num_frames / 2) <= 11:
+        return ratio
+    frame_id = 0
+    while frame_id in range(num_frames):
+        start = random.choice(range(min(10, num_frames), min(int(num_frames / 2), 70)))
+        if frame_id + start + 5 <= num_frames - 1:
+            ratio[frame_id + start : frame_id + start + 5, 0] = [
+                0.5,
+                0.9,
+                1.0,
+                0.9,
+                0.5,
+            ]
+            frame_id = frame_id + start + 5
+        else:
+            break
+    return ratio
+
+
+# so we patch in a fixed version
+gb.generate_blink_seq_randomly = fixed_generate_blink_seq_randomly
+
+
+def urlretrieve(url, filename):
+    """Same as urllib.urlretrieve but uses requests because urllib breaks on discord attachments. Does not support data: URLs and local files."""
+    res = requests.get(url)
+    if not res.ok:
+        raise ValueError(
```

Comment:
This should use our existing methods to handle bad user URLs ie raise_for_status

Reply:
[bc840b2](https://github.com/GooeyAI/gooey-gpu/pull/26/commits/bc840b28cde5386fb4fb5b976102329a70070796)